apps/src/lab2/views/Lab2Wrapper.tsx

```diff
@@ -28,6 +31,16 @@ const Lab2Wrapper: React.FunctionComponent<Lab2WrapperProps> = ({children}) => {
     (state: {lab: LabState}) =>
       state.lab.pageError?.errorMessage || state.lab.pageError?.error?.message
   );
+  const dispatch = useAppDispatch();
+
+  // Store the level ID provided by App Options in redux. This is needed
+  // on pages without a header, such as the share view.
+  const serverLevelId = getServerLevelId();
+  useEffect(() => {
+    if (serverLevelId) {
+      dispatch(setCurrentLevelId(serverLevelId.toString()));
```

Comment:
This is effectively a replacement for https://github.com/code-dot-org/code-dot-org/blob/8cf1b3b09edb95051da918d2db669bd81fa8aa15/apps/src/code-studio/header.js#L140 which is not called here because there's no header in the share view.